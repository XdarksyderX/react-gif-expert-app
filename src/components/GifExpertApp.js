import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

export const GifExpertApp = () => {
	const [categories, setCategories] = useState(['One Punch']);

	return (
		<div className="app">
			<h2>GifExpertApp</h2>
			
			<AddCategory setCategories={setCategories}/>
            <ul>
                {categories.map(cat => <GifGrid key={cat} category={cat}/>)}
            </ul>
			<hr />
		</div>
	);
};
