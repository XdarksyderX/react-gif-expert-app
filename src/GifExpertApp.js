import React from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = (  ) => {
	const [categories, setCategories] = React.useState([]);

	return (
		<div className="app">
			<h2>GifExpertApp</h2>
			
			<AddCategory setCategories={setCategories}/>
            <ul>
                {categories.map(cat => <GifGrid key={cat} category={cat}/>)}
            </ul>
			<hr />
		</div>
	);
};
