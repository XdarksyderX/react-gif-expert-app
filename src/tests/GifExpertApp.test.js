import React from 'react'
import {shallow} from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';


describe('GifExpertApp: ', () => {
    

    test('Should render correctly', () => {
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('Should display a categories list', () => {
        const initialStateCategories = ['One Punch Man', 'Dragon Ball'];
        const realUseState = React.useState;


        jest.spyOn(React, 'useState').mockImplementationOnce(() => realUseState(initialStateCategories))

        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(initialStateCategories.length);
    })
});