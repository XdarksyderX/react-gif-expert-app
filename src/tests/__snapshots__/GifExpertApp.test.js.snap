// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp:  Should display a categories list 1`] = `ShallowWrapper {}`;

exports[`GifExpertApp:  Should render correctly 1`] = `ShallowWrapper {}`;
